### PR TITLE
migmap wrapper with test

### LINKED ISSUE #1493: WIP new wrapper for migmap
I'm working on a wrapper for [migmap](https://github.com/mikessh/migmap) with the intention of using it to classify results from the presto tools that are also in process #1454

### PR DESCRIPTION
**Pull request by bwlang**
fixes #1493

### CODE REVIEW

**Comment by mvdbeek** [comment]
@bwlang can you add a `.shed.yml` file ? This is necessary for the tool tests to run on travis, and we're recommending single quotes around paths. Also the `settings` conditional doesn't seem to do anything (yet?).

File: tools/migmap/migmap.xml [hunk 1]
```diff
@@ -0,0 +1,161 @@
+<tool id="migmap" name="MiGMAP: mapper for full-length T- and B-cell repertoire sequencing" version="1.0.2.1">
+    <description>This software is a smart wrapper for IgBlast V-(D)-J mapping tool designed to facilitate analysis immune receptor libraries profiled using high-throughput sequencing</description>
+    <requirements>
+	 <requirement type="package" version="1.0.2">migmap</requirement>	
+    </requirements>
+    <stdio>
```
Comment (bgruening) [inline tools/migmap/migmap.xml]: using detect_code you can remove the stdio

File: tools/migmap/migmap.xml [hunk 2]
```diff
@@ -0,0 +1,161 @@
+<tool id="migmap" name="MiGMAP: mapper for full-length T- and B-cell repertoire sequencing" version="1.0.2.1">
+    <description>This software is a smart wrapper for IgBlast V-(D)-J mapping tool designed to facilitate analysis immune receptor libraries profiled using high-throughput sequencing</description>
```
Comment (bgruening) [inline tools/migmap/migmap.xml]: This description should probably go into the help. You want to have the name- and the description short as this pops up in your tool panel.

File: tools/migmap/migmap.xml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+         "$output"
+    ]]></command>
+    <inputs>
+        <param type="data" name="input" format="fasta,fastqsanger" />
+        <param type="select" name="species" label="Species" value='human' >
+		<option value='human' selected='true'>Human</option>
+		<option value='mouse'>Mouse</option>
+                <option value='rat'>Rat</option>
+                <option value='rabbit'>Rabbit</option>
+	        <option value='rhesus_monkey'>Rhesus monkey</option>
```
Comment (bgruening) [inline tools/migmap/migmap.xml]: indentation

File: tools/migmap/migmap.xml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+         $all_alleles 
+         #if str($qual_threshold) :
+         -q $qual_threshold
+         #end if
+         -p "\${GALAXY_SLOTS:-4}"
+         --data-dir "\${IGBLAST_PATH}/../share/igblast"
+         -S $species 
+         -R $receptor_list
+         --report "$report" 
+         "$input"
```
Comment (bgruening) [inline tools/migmap/migmap.xml]: as @mvdbeek already pointed out, please use single-quotes if you don't need variable expansion.

File: tools/migmap/migmap.xml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                truevalue="--allow-incomplete" falsevalue="" />
+        <param name="all_alleles" type="boolean" label="use all alleles during alignemt (default = only major (*01) alleles)" checked='false' 
+                truevalue="--all-alleles" falsevalue="" />
+        <param name="allow_no_cdr3" type="boolean" label="Report clonotypes with no CDR3" checked='false' 
+                truevalue="--allow-no-cdr3" falsevalue="" />
+        <param name="allow_noncanonical" type="boolean" label="Report clonotypes that have non-canonical CDR3" checked='false' 
+                truevalue="--allow-noncanonical" falsevalue="" />
+        <param name="allow_noncoding" type="boolean" label="Report non-coding clonotypes (stop codon or frameshift)" checked='false' 
+                truevalue="--allow-noncoding" falsevalue="" />
+        <param name="by_read" type="boolean" label="Report mapping details for each read" checked='false'
```
Comment (bgruening) [inline tools/migmap/migmap.xml]: This is not used in the command section?

File: tools/migmap/migmap.xml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+            <output name="report" file="test_stop_codon.report"/> 
+            <output name="output" file="test_stop_codon.out"/>
+        </test>
+        <test>
+            <param name="input" value="test_out_of_frame.fa"/>
+            <output name="report" file="test_out_of_frame.report"/> 
+            <output name="output" file="test_out_of_frame.out"/>
+        </test>
+    </tests>
+    <help><![CDATA[
```
Comment (bgruening) [inline tools/migmap/migmap.xml]: the arguments are not typically useful for the users, a short description of the tool, links to upstream documentation should be nicer, imho

**Comment by bwlang** [comment]
@bgruening @mvdbeek  
thanks for the feedback... I think i've incorporated all your suggestions and the tests are now running and passing - yea!

**Comment by bgruening** [comment]
:+1:

**Comment by bgruening** [comment]
Thanks @bwlang!

**Pull request merged**